Worker/src/Common/Archive/ZipArchiver.cs

```diff
@@ -280,4 +189,132 @@ public static string GetLocalPathToAssembly(string pathToZip)
 
     return pathToAssembly;
   }
+
+  /// <inheritdoc />
+  string IArchiver.DownloadArchive(IFileAdapter fileAdapter,
+                                   string       fileName,
+                                   bool         skipIfExists)
+  {
+    if (!skipIfExists || !File.Exists(Path.Combine(fileAdapter.DestinationDirPath,
+                                                   fileName)))
+    {
+      return fileAdapter.DownloadFile(fileName);
+    }
+
+    return Path.Combine(fileAdapter.DestinationDirPath,
+                        fileName);
+  }
+
+  /// <summary>
+  /// </summary>
+  /// <param name="assemblyNameFilePath"></param>
+  /// <returns></returns>
+  public static bool IsZipFile(string assemblyNameFilePath)
+  {
+    //ATm ONLY Check the extensions 
+
+    var extension = Path.GetExtension(assemblyNameFilePath);
+    return extension?.ToLower() == ".zip";
+  }
+
+  /// <summary>
+  /// </summary>
+  /// <param name="assemblyNameFilePath"></param>
+  /// <returns></returns>
+  /// <exception cref="WorkerApiException"></exception>
+  public static IEnumerable<string> ExtractNameAndVersion(string assemblyNameFilePath)
+  {
+    string filePathNoExt;
+    string appName;
+    string versionName;
+
+    try
+    {
+      filePathNoExt = Path.GetFileNameWithoutExtension(assemblyNameFilePath);
+    }
+    catch (ArgumentException e)
+    {
+      throw new WorkerApiException(e);
+    }
+
+    // Instantiate the regular expression object.
+    var pat = @"(.*)-v(.+)";
```

Comment:
Could you reimplement this based on split? It would be simpler, faster, and more robust. You would be able to easily detect if there are multiple `-v`in the string.